### PR TITLE
Commerce 2525 1

### PR DESCRIPTION
**Pull request by riccardo-ferrari**

### CODE REVIEW

**Comment by riccardo-ferrari** [comment]
ci:test

**Comment by marco-leo** [comment]
ci:test

**Comment by riccardo-ferrari** [comment]
ci:test

**Comment by riccardo-ferrari** [comment]
ci:test

**Comment by marco-leo** [comment]
@ethanbustad looks good to me, please ask changes if you see it needs.  Thanks

File: commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDisplayLayoutLocalServiceImpl.java [hunk 1]
```diff
@@ -114,12 +116,25 @@ public CPDisplayLayout deleteCPDisplayLayout(Class<?> clazz, long classPK) {
 		return null;
 	}
 
+	@Override
+	public void deleteCPDisplayLayoutByG_L(long groupId, String layoutUuid) {
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDisplayLayoutLocalServiceImpl.java:120]: We shouldn't use the `*ByG_L` as part of the public name here -- we always make this friendlier. It looks like we could just call it `deleteCPDisplayLayout` (as long as that signature with `(long, String)` isn't already taken). If not, we'd probably call it `deleteCPDisplayLayoutByLayoutUuid` (or maybe `ByLayout` to make it even a little friendlier).

File: commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDisplayLayoutLocalServiceImpl.java [hunk 2]
```diff
@@ -114,12 +116,25 @@ public CPDisplayLayout deleteCPDisplayLayout(Class<?> clazz, long classPK) {
...
+	}
+
 	@Override
 	public CPDisplayLayout fetchCPDisplayLayout(Class<?> clazz, long classPK) {
 		return cpDisplayLayoutPersistence.fetchByC_C(
 			classNameLocalService.getClassNameId(clazz), classPK);
 	}
 
+	@Override
+	public List<CPDisplayLayout> fetchCPDisplayLayoutByG_L(
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDisplayLayoutLocalServiceImpl.java:131]: same here

File: commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+		_layoutLocalService = layoutLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		Statement s = connection.createStatement(
+			ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
+
+		ResultSet r = s.executeQuery(_SELECT_CPDISPLAYLAYOUT_SQL);
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java:42]: These should be closed in a `finally` -- or better yet, automatically by a try-with-resources.

File: commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+	private Layout _getLayout(long groupId, String layoutUuid) {
+		Layout layout = _layoutLocalService.fetchLayout(
+			layoutUuid, groupId, false);
+
+		if (layout != null) {
+			return layout;
+		}
+
+		return _layoutLocalService.fetchLayout(layoutUuid, groupId, false);
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java:76]: shouldn't it be `(layoutUuid, groupId, true)`?

File: commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+				if (_log.isDebugEnabled()) {
+					_log.debug(
+						String.format(
+							"Removing orphan CPDisplayLayout entry id: %s",
+							cpDisplayLayoutId));
+				}
+
+				runSQL(
+					String.format(
+						_DELETE_CPDISPLAYLAYOUT_SQL, cpDisplayLayoutId));
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java:63]: This..... is wrong, isn't it?
`"DELETE FROM CPDisplayLayout where '%s'".format(id)`
yields something like
`"DELETE FROM CPDisplayLayout where '1234'"`.

We need to name the column we're `where`ing by, and I don't think we should be quoting the ID, either.

File: commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+			}
+		}
+	}
+
+	private Layout _getLayout(long groupId, String layoutUuid) {
```
Comment (ethanbustad) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java:68]: should probably call this `_fetchLayout`, just to be as clear as possible that the result could very well be `null`.

**Review by ethanbustad — COMMENTED** [review]
@riccardo-ferrari take a look, I think some of these changes could be pretty important

File: commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java (hunk 5 above)
Comment (riccardo-ferrari) [reply commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_1/CPDisplayLayoutUpgradeProcess.java:63]: ouch this was bad, I guess I should get some rest

**Pull request merged**